### PR TITLE
fix accessReview integration tests

### PR DESCRIPTION
**Pull request by deads2k**
Fixes https://github.com/openshift/origin/issues/2899

Turns out it was just bad tests.  The empty namespace used for the cluster requests works fine.

@liggitt

### CODE REVIEW

**Comment by deads2k** [comment]
now with upstream patch.

**Comment by liggitt** [comment]
[test]

File: test/integration/authorization_test.go [hunk 1]
```diff
@@ -159,7 +159,7 @@ func (test resourceAccessReviewTest) run(t *testing.T) {
 		}
 	}
 
-	if reflect.DeepEqual(actualResponse, test.response) {
+	if !reflect.DeepEqual(actualResponse.Users.List(), test.response.Users.List()) || !reflect.DeepEqual(actualResponse.Groups.List(), test.response.Groups.List()) {
```
Comment (liggitt) [inline test/integration/authorization_test.go]: can you do the message prefix compare here as well?

Comment (deads2k) [inline test/integration/authorization_test.go]: > can you do the message prefix compare here as well?

Added namespace.  There is no message.

**Comment by liggitt** [comment]
```
--- FAIL: TestBootstrapPolicySelfSubjectAccessReviews-2 (7.66s)
	authorization_test.go:295: : from review
			&api.SubjectAccessReview{TypeMeta:api.TypeMeta{Kind:"", APIVersion:""}, Verb:"create", Resource:"policybindings", User:"", Groups:util.StringSet(nil), Content:runtime.EmbeddedObject{Object:runtime.Object(nil)}, ResourceName:""}
		expected
			&api.SubjectAccessReviewResponse{TypeMeta:api.TypeMeta{Kind:"", APIVersion:""}, Namespace:"openshift", Allowed:false, Reason:"denied by default"}
		got
			&api.SubjectAccessReviewResponse{TypeMeta:api.TypeMeta{Kind:"", APIVersion:""}, Namespace:"openshift", Allowed:false, Reason:"User \"valerie\" cannot create policybindings in project \"openshift\""}
	authorization_test.go:284: : expected
			cannot 
		got
			User "valerie" cannot create subjectaccessreviews in project "openshift"
FAIL
```

**Comment by liggitt** [comment]
sorry, I didn't realize other files were using the subjectAccessReviewTest

File: Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver/handlers.go [hunk 2]
```diff
@@ -340,19 +340,6 @@ func (r *APIRequestInfoResolver) GetAPIRequestInfo(req *http.Request) (APIReques
 				currentParts = currentParts[2:]
 			}
 		}
-	} else {
```
Comment (liggitt) [inline Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver/handlers.go]: I kept the else and set to this in my upstream PR:
```
requestInfo.Namespace = api.NamespaceAll
```

**Comment by liggitt** [comment]
Upstream PR at https://github.com/GoogleCloudPlatform/kubernetes/pull/9600

**Comment by liggitt** [comment]
With those two chunks removed, I got a clean unit and integration test pass with upstream. Once we get a clean test run, this should go in. @smarterclayton approve?

**Comment by smarterclayton** [comment]
```
11e5-a3d5-22000b3b0557", Groups:[]string(nil)}
--- FAIL: TestBootstrapPolicySelfSubjectAccessReviews-2 (7.66s)
	authorization_test.go:295: : from review
			&api.SubjectAccessReview{TypeMeta:api.TypeMeta{Kind:"", APIVersion:""}, Verb:"create", Resource:"policybindings", User:"", Groups:util.StringSet(nil), Content:runtime.EmbeddedObject{Object:runtime.Object(nil)}, ResourceName:""}
		expected
			&api.SubjectAccessReviewResponse{TypeMeta:api.TypeMeta{Kind:"", APIVersion:""}, Namespace:"openshift", Allowed:false, Reason:"denied by default"}
		got
			&api.SubjectAccessReviewResponse{TypeMeta:api.TypeMeta{Kind:"", APIVersion:""}, Namespace:"openshift", Allowed:false, Reason:"User \"valerie\" cannot create policybindings in project \"openshift\""}
	authorization_test.go:284: : expected
			cannot 
		got
			User "valerie" cannot create subjectaccessreviews in project "openshift"
FAIL
```

**Comment by liggitt** [comment]
right, hasn't been updated since https://github.com/openshift/origin/pull/3044#issuecomment-110905988

**Comment by liggitt** [comment]
opened #3082 with fix for failing test

**Comment by deads2k** [comment]
@liggitt's update merged.

**Pull request closed without merging**